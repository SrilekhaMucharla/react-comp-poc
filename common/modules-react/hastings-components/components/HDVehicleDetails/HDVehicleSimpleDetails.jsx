import React from 'react';
import PropTypes from 'prop-types';
import {
    Row, Col
} from 'react-bootstrap';

const HDVehicleSimpleDetails = ({ vrn, displayName, className }) => {
    return (
        <Row className={`vehicle_simple_details_row ${className}`}>
            <Col className="vehicle_simple_details_col">
                <div className="vrn_highlighted">
                    <span>{vrn}</span>
                </div>
                <div className="vheicle_detail_display_name_holder">
                    <span className="vheicle_detail_display_name">{displayName}</span>
                </div>
            </Col>
        </Row>
    );
};

HDVehicleSimpleDetails.propTypes = {
    vrn: PropTypes.string.isRequired,
    displayName: PropTypes.string.isRequired,
    className: PropTypes.string,
};

HDVehicleSimpleDetails.defaultProps = {
    className: ''
};


export default React.memo(HDVehicleSimpleDetails);
