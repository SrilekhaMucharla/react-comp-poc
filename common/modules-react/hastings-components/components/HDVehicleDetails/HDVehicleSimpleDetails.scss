.vehicle_simple_details_row {
    margin-top:28px;
    .vehicle_simple_details_col {
        display:flex;
        flex-direction: row;
        align-items: center;
        .vheicle_detail_display_name_holder {
        .vheicle_detail_display_name {
            font-size: 14px;
            font-weight: 500;
            letter-spacing: normal;
            text-align: left;
            color:$mono-midnight;
        }
    }
        .vrn_highlighted {
            > span {
                padding: 3px;
                background-color: #fbe700;
                margin-right: 13px;
                line-height: 1.35;
                font-family: $font-licence-plate;
                font-stretch: normal;
                font-style: normal;
                text-align: left;
                font-size: 13px;
            }
        }
    }
}