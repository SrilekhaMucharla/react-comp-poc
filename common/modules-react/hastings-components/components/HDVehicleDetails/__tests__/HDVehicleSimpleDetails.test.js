import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import HDVehicleSimpleDetails from '../HDVehicleSimpleDetails';

configure({ adapter: new Adapter() });

describe('<HDVehicleSimpleDetails/>', () => {
    it('render component with vrn and display name', () => {
        const vrn = 'AV12 BGE';
        const displayName = 'MERCEDES-BENZ E250 SPORT ED125 CDI BLUE';

        const tree = renderer.create(<HDVehicleSimpleDetails
            vrn={vrn}
            displayName={displayName} />)
            .toJSON();

        expect(tree)
            .toMatchSnapshot();
    });
});
