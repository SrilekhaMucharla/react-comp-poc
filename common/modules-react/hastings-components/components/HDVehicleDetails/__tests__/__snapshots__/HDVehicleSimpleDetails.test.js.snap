// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDVehicleSimpleDetails/> render component with vrn and display name 1`] = `
<div
  className="vehicle_simple_details_row  row"
>
  <div
    className="vehicle_simple_details_col col"
  >
    <div
      className="vrn_highlighted"
    >
      <span>
        AV12 BGE
      </span>
    </div>
    <div
      className="vheicle_detail_display_name_holder"
    >
      <span
        className="vheicle_detail_display_name"
      >
        MERCEDES-BENZ E250 SPORT ED125 CDI BLUE
      </span>
    </div>
  </div>
</div>
`;
